common/rya.api/src/main/java/org/apache/rya/api/functions/DateTimeWithinPeriod.java

```diff
@@ -0,0 +1,121 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */package org.apache.rya.api.functions;
+
+import org.apache.rya.api.functions.OWLTime.Duration;
+import org.openrdf.model.Literal;
+import org.openrdf.model.URI;
+import org.openrdf.model.Value;
+import org.openrdf.model.ValueFactory;
+import org.openrdf.model.datatypes.XMLDatatypeUtil;
+import org.openrdf.model.vocabulary.FN;
+import org.openrdf.model.vocabulary.XMLSchema;
+import org.openrdf.query.algebra.evaluation.ValueExprEvaluationException;
+import org.openrdf.query.algebra.evaluation.function.Function;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * This {@link Function} determines whether two datetimes occur within a specified period of time of one another. The
+ * method {@link Function#evaluate(ValueFactory, Value...)} expects four values, where the first two values are the
+ * datetimes, the third value is an integer indicating the period, and the fourth value is a URI indicating the time
+ * unit of the period. The URI must be of Type DurationDescription in the OWL-Time ontology (see
+ * <a href ="https://www.w3.org/TR/owl-time/">https://www.w3.org/TR/owl-time/</a>). Examples of valid time unit URIs
+ * can be found in the class {@link OWLTime} and below
+ * <ul>
+ * <li>http://www.w3.org/2006/time#days</li>
+ * <li>http://www.w3.org/2006/time#hours</li>
+ * <li>http://www.w3.org/2006/time#minutes</li>
+ * <li>http://www.w3.org/2006/time#seconds</li>
+ * </ul>
+ * 
+ */
+public class DateTimeWithinPeriod implements Function {
+
+    private static final String DATE_TIME_WITHIN = "dateTimeWithin";
+
+    @Override
+    public String getURI() {
+        return FN.NAMESPACE.toString() + DATE_TIME_WITHIN;
+    }
+
+    /**
+     * Determines whether two datetimes occur within a specified period of time of one another. This method expects four
+     * values, where the first two values are the datetimes, the third value is an integer indicating the period, and
+     * the fourth value is a URI indicating the time unit of the period. The URI must be of Type DurationDescription in
+     * the OWL-Time ontology (see <a href ="https://www.w3.org/TR/owl-time/">https://www.w3.org/TR/owl-time/</a>).
+     * Examples of valid time unit URIs can be found in the class {@link OWLTime} and below
+     * <ul>
+     * <li>http://www.w3.org/2006/time#days</li>
+     * <li>http://www.w3.org/2006/time#hours</li>
+     * <li>http://www.w3.org/2006/time#minutes</li>
+     * <li>http://www.w3.org/2006/time#seconds</li>
+     * </ul>
+     * 
+     * @param valueFactory - factory for creating values
+     * @param values - array of Value arguments for this Function.
+     */
+    @Override
+    public Value evaluate(ValueFactory valueFactory, Value... values) throws ValueExprEvaluationException {
+
+        // general validation of input
+        Preconditions.checkArgument(values.length == 4);
+        Preconditions.checkArgument(values[0] instanceof Literal);
```

Comment:
This is a style comment. It would be a little easier to understand that this code is doing if these array indices were descriptive of what was stored in the array at that position.

for exmple:
private static final int DATE_TIME_1 = 0;
private static final int DATE_TIME_2 = 1;
private static final int PERIOD = 2;
private static final int PERIOD_UNITS = 3;

Then the rest of the code could look like:
long epochTime1 = convertDatetimeToMillis((Literal) values[DATE_TIME_1]);
long epochTime2 = convertDatetimeToMillis((Literal) values[DATE_TIME_2]);
long periodMillis = convertPeriodToMillis((Literal) values[PERIOD], (URI) values[PERIOD_UNITS]);

etc

Reply:
Point taken.  I think the comments clearly explain the order of the arguments.